tomviz/modules/ModuleManager.cxx

```diff
@@ -724,6 +736,21 @@ bool ModuleManager::deserialize(const QJsonObject& doc, const QDir& stateDir)
     vtkSMPropertyHelper(viewProxy, "EyeAngle")
       .Set(camera["EyeAngle"].toDouble());
     viewProxy->UpdateVTKObjects();
+    if (view.contains("scaleLegend")) {
+      auto scaleLegend = ScaleLegend::getScaleLegend(viewProxy);
+      if (scaleLegend) {
+        auto legendType = view["scaleLegend"].toString();
+        if (legendType == "none") {
+          scaleLegend->setVisibility(false);
+        } else if (legendType == "cube") {
+          scaleLegend->setVisibility(true);
+          scaleLegend->setStyle(ScaleLegendStyle::Cube);
+        } else if (legendType == "ruler") {
```

Comment:
This shouldn't happen, but we should probably favor a default, and perhaps issue a warning. This is not enough to correctly update the view menu with the state.